From e8281e3cbb88ba322c5c920fb6e30c8f734e1bee Mon Sep 17 00:00:00 2001
From: Chester Lin <clin@suse.com>
Date: Sat, 12 Oct 2019 03:01:38 +0200
Subject: [PATCH] net: bcmgenet: decrease-dma-burst-size
Patch-Mainline: Not yet, a temp workaround
References: rpi4 porting

A temporarily patch to workaround genet transmission timed-out issue.

This patch refers to Matthias Brugger's work:
series: This series adds ethernet support for RPi4
link: https://patchwork.kernel.org/project/linux-arm-kernel/list/?series=186389

Signed-off-by: Matthias Brugger <mbrugger@suse.com>
Signed-off-by: Chester Lin <clin@suse.com>
---
 drivers/net/ethernet/broadcom/genet/bcmgenet.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/broadcom/genet/bcmgenet.h
+++ b/drivers/net/ethernet/broadcom/genet/bcmgenet.h
@@ -28,7 +28,7 @@
 #define ENET_PAD		8
 #define ENET_MAX_MTU_SIZE	(ETH_DATA_LEN + ETH_HLEN + VLAN_HLEN + \
 				 ENET_BRCM_TAG_LEN + ETH_FCS_LEN + ENET_PAD)
-#define DMA_MAX_BURST_LENGTH    0x10
+#define DMA_MAX_BURST_LENGTH    0x08
 
 /* misc. configuration */
 #define CLEAR_ALL_HFB			0xFF
